cmd/sidecars/network-passt-binding/domain/configurator.go

```diff
@@ -95,6 +95,25 @@ func (p PasstNetworkConfigurator) Mutate(domainSpec *domainschema.DomainSpec) (*
 	} else {
 		domainSpecCopy.Devices.Interfaces = append(domainSpecCopy.Devices.Interfaces, *generatedIface)
 	}
+	const (
+		sharedMemoryBackingAccessMode = "shared"
+		memfdMemoryBackingSourceType  = "memfd"
+	)
+	if domainSpecCopy.MemoryBacking == nil {
+		domainSpecCopy.MemoryBacking = &domainschema.MemoryBacking{
+			Access: &domainschema.MemoryBackingAccess{
+				Mode: sharedMemoryBackingAccessMode,
+			},
+			Source: &domainschema.MemoryBackingSource{
+				Type: memfdMemoryBackingSourceType,
+			},
+		}
+	} else {
+		if domainSpecCopy.MemoryBacking.Access != nil && domainSpecCopy.MemoryBacking.Access.Mode != sharedMemoryBackingAccessMode {
```

Comment:
do we need to also verify that `MemoryBacking.Source.Type` has the desired state ?